test/pylib/scylla_cluster.py

```diff
@@ -488,6 +489,7 @@ def _seeds(self) -> List[str]:
     async def add_server(self) -> str:
         """Add a new server to the cluster"""
         server = self.create_server(self.name, self._seeds())
+        self.is_dirty = True
```

Comment:
I have some doubts about the concept of "dirty". What makes a cluster "dirty"? Is a cluster which started with 3 nodes, one was added and then one was removed, "dirty"? Why?

Reply:
I also requested a comment explaining what `is_dirty` means, among some other fields, I see some of the ones I requested were commented upon but not all, including `is_dirty`:
https://github.com/scylladb/scylladb/pull/10815#discussion_r935629490

> Is a cluster which started with 3 nodes, one was added and then one was removed, "dirty"? Why?

I guess any topology change makes a cluster dirty. We definitely want to mark cluster dirty if it's missing a node, for example. In a follow-up we may think about optimizations (but let's not do this right now), e.g. maybe if the number of nodes is the same before and after we can leave it "not dirty" (but I'm not sure).

Reply:
Actually, `is_dirty` is commented upon now:
```
        # cluster was modified in a way it should not be used in subsequent tests
        self.is_dirty: bool = False
```

The discussion what modifications should make a cluster dirty... We could expand on it in the comments, but meh, I don't think it matters that much.

We can improve on this later.

Reply:
The reason I asked for a comment is that it's not clear what "cluster was modified" really means. Here is a concrete example that led me to this question: A lot of existing dtests create a three-node cluster, then kill one node, and do something, and then bring back the third node to life and test what happened. So when the test has ended, we again have a working three-node cluster. Is such a cluster "dirty" or not "dirty"? Should it be used in subsequent tests, or not?

But I can accept that for now, we don't know exactly what "is_dirty" will be - and that it will continue to evolve in the near future. If this is the case, then indeed the comment we have is good enough.